core/math/vector2.h

```diff
@@ -114,6 +114,25 @@ struct _NO_DISCARD_ Vector2 {
 	_FORCE_INLINE_ Vector2 bezier_interpolate(const Vector2 &p_control_1, const Vector2 &p_control_2, const Vector2 &p_end, const real_t p_t) const;
 	_FORCE_INLINE_ Vector2 bezier_derivative(const Vector2 &p_control_1, const Vector2 &p_control_2, const Vector2 &p_end, const real_t p_t) const;
 
+	_FORCE_INLINE_ Vector2 rotate_toward(Vector2 p_to, real_t p_delta) const {
+		if (p_delta < 0.0f) {
+			p_delta = -p_delta;
+			p_to = -p_to;
+		}
+
+		real_t angle = Math::abs(angle_to(p_to));
+
+		if (angle < p_delta) {
+			if (unlikely(p_to.length_squared() == 0.0f)) {
+				// Prevent locking up when p_to is (0, 0).
+				return lerp(p_to, p_delta);
+			}
+			return move_toward(p_to, p_delta * p_to.length());
+		}
```

Comment:
This is why I was confused; it looks like radian and length are mixed up. Isn't it strange that the delta of rotate_toward is radian, yet it is converted to length? If the behavior is to change the length with priority on the angle, shouldn't this already be the destination vector?

```suggestion
		if (angle <= p_delta) {
			return p_to;
		}
```

Reply:
The API did originally have separate controls for movement (length) and rotation, which is why it ended up being so convoluted. Keep length was originally proposed but I think it wasn't went with because it's relatively easy to replicate `v1.normalized().rotate_toward(v2.normalized(), delta) * v1.length()`, but I agree it's a nice convenience feature.